### PR TITLE
fix: uploading appData too late

### PR DESCRIPTION
**Pull request by compojoom**
# Summary

We were trying to decode the TWAP transactions that our users were doing. It turned out our decoding was working for some, but not for other txs. To present a good decoding of the TWAP transaction we need the appData as it contains information for the slippage, fee etc. It turned out that some of the txs were not having app data. This is due to the fact that we are trying to decode the transaction before signing, but at this point no appData was being uploaded to the server.

Moving the uploadAppData call before the call for signing the tx solves this problem and makes sure that appData is available even before signing.

# To Test
Open the network tab, create a twap order, hit the "place twap order" button. You should see that the app_data is being uploaded even before there is a signature from the user.

### CODE REVIEW

**Review by alfetopito — APPROVED** [review]
I have the concern of polluting the db with appData that's never used, as it's not likely all signature preview attempts will result in an order being placed.
However, this is not different to how non-TWAP orders (pre-sign) are placed, and the backend is fine with it.

**Review by shoom3301 — APPROVED** [review]
Thank you!

File: apps/cowswap-frontend/src/modules/twap/hooks/useCreateTwapOrder.tsx [hunk 1]
```diff
@@ -97,6 +97,9 @@ export function useCreateTwapOrder() {
         const fallbackSetupTxs = fallbackHandlerIsNotSet
           ? await extensibleFallbackSetupTxs(extensibleFallbackContext)
           : []
+
+        // upload the app data here, as application might need it to decode the order info before it is being signed
+        uploadAppData({ chainId, orderId, appData: appDataInfo })
         const createOrderTxs = createTwapOrderTxs(twapOrder, paramsStruct, twapOrderCreationContext)
         const { safeTxHash } = await safeAppsSdk.txs.send({ txs: [...fallbackSetupTxs, ...createOrderTxs] })
```
Comment (anxolin) [inline apps/cowswap-frontend/src/modules/twap/hooks/useCreateTwapOrder.tsx:104]: nit: we could consider deleting it if this `txs.send(...)` fails, right? (i.e. user rejects the transaction). Maybe this helps with the pollution?

**Review by anxolin — APPROVED** [review]

**Review by elena-zh — APPROVED** [review]

**Comment by elena-zh** [comment]
Hey @compojoom , thank you!
I was not able to test these changes in the widget, however, on CoW Swap it works as described
widget

**Pull request merged**